File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        ans = []
        
        for i in range(len(nums)):
            if nums[i] < pivot:
                ans.append(nums[i])
        
        for j in range(len(nums)):
            if nums[j] == pivot:
                ans.append(nums[j])
        
        for k in range(len(nums)):
            if nums[k] > pivot:
                ans.append(nums[k])
        return ans

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        List<Integer> left_side = new ArrayList<>();
        List<Integer> equals = new ArrayList<>();
        List<Integer> right_side = new ArrayList<>();
        
        for (int i=0; i < nums.length; i++){
            if (nums[i] < pivot) left_side.add(nums[i]);
            else if (nums[i] > pivot) right_side.add(nums[i]);
            else equals.add(nums[i]);
        }

        left_side.addAll(equals);
        left_side.addAll(right_side);
        
        int[] answer = left_side.stream().mapToInt(i -> i).toArray();
        return answer;

    }
}